woocommerce/woocommerce
# Refactoring Task Infrastructure

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:



This pull request makes a number of changes to our task orchestration infrastructure in order to introduce consistency and significantly improve cache performance. Generally speaking, from the user-perspective, there should be very little changed by this pull request.

**Note: I chose to batch the changes in this PR so that it will have a minimal impact on developer experience by being merged at the same time.**



Closes https://github.com/woocommerce/woocommerce/issues/40954, https://github.com/woocommerce/woocommerce/issues/40952, https://github.com/woocommerce/woocommerce/issues/40953.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Verify that CI is passing. All of the changes in this PR apply to CI as well.
2. `pnpm --filter='@woocommerce/plugin-woocommerce' build`
3. `pnpm --filter='@woocommerce/plugin-woocommerce' lint`
4. `pnpm --filter='@woocommerce/plugin-woocommerce' build`
5. Explore! Play around with different commands and see how things work out.



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
